### DISCUSSION TITLE
iOS messaging().getToken() is not working in some devices

### DISCUSSION THREAD

**Discussion by mantmartini** [invertase/react-native-firebase#6236] [Q&A]
The problem is that in some cases, my app doesn't get the device token when calling messaging().getToken(). I don't know the reasons. This only happens in ios.

On the other hand, on Android, it seems this works correctly. So, I think this problem is caused by some permissions in iphones devices, could be? 

Otherwise, the current version of firebase library in my project is 11.1.0. Is too old? Can it cause this problem? 
Should I updgrade the firebase library?

**Comment by mikehardy** [comment]
Oh wow, 11.1.0. Is too old yes. It's from ...about 15 months ago. You're missing nearly a year and a half of fixes.

Never attempt to troubleshoot or debug on old versions, since you will frequently be chasing ghosts (issues that are already fixed...).

There was a *lot* of work around token fetching and I don't think any of your experimental results will be valid anymore but I also think all the fetching issues are fixed, at least I really hope so - I expect you'll see positive results with current versions.

You'll be crossing quite a few major versions but if you read the CHANGELOG carefully I really try my best to outline *exactly* what to change and even exactly how if possible -

https://invertase.io/blog/react-native-firebase-versioning
https://github.com/invertase/react-native-firebase/blob/main/CHANGELOG.md (minors and patches should "just work", just read the BREAKING CHANGES notes on the major versions)

Hopefully this helps - only hangup I could think is if you are/were using admob you'll want to migrate to our companion project https://github.com/invertase/react-native-google-mobile-ads/blob/main/CHANGELOG.md which is where we moved those APIs after firebase removed them upstream

Cheers
